### ISSUE TITLE
optionally create a router when deploying the controller

### ISSUE THREAD

**Issue by qrkourier** [openziti/helm-charts#108]
It's highly likely that a controller deployment is immediately followed by a router deployment. We can simplify and facilitate this by automatically provisioning a router in the Ziti mgmt API and storing the token as a K8s secret. 

The implementation will involve Helm hooks and simple shell scripts to be executed by the resulting Job resources that are scheduled during life cycle events.

**Comment by qrkourier** [comment]
The point of this issue is to simplify the process of getting a Ziti network running in K8s by eliminating these orchestration steps:

1. fetch Ziti admin password
2. login to Ziti
3. create router and save token
4. fetch token
5. supply token to router chart

**Comment by dariuszSki** [comment]
yes, if the router will be deployed on the same cluster, then I would make sense to add an option to create a router if needed

**Comment by qrkourier** [comment]
I'm still planning to work on this soon.

**Comment by qrkourier** [comment]
I'm clawing my way back to this and still think it's an obvious way to simplify k8s bootstrapping.

**Comment by qrkourier** [comment]
This avoids Terraforming/Ansible the mgmt API to create the router and is relevant to [the "ziti-stack" chart idea](https://github.com/openziti/helm-charts/issues/216) (umbrella chart for controller+cert-manager+trust-manager+router w/ optional node proxy daemonset, sidecar injector, etc.)

**Comment by qrkourier** [comment]
This will be handled by the Operator
